userguide/storage.rst

```diff
@@ -713,53 +713,70 @@ existing datasets to close them--the remaining
-   from :ref:`Shell`. However, any data that has already been
-   deduplicated will not be un-deduplicated. Only newly stored data
-   after the property change will not be deduplicated. The only way to
-   remove existing deduplicated data is to copy all of the data off of
-   the dataset, set the property to off, then copy the data back in
-   again. Alternately, create a new dataset with
-   :guilabel:`ZFS Deduplication` left disabled, copy the data to the
-   new dataset, and destroy the original dataset.
+#ifdef freenas
+.. index:: Deduplication
+.. _Deduplication:
+
+Deduplication
+^^^^^^^^^^^^^
+
+Deduplication ("dedup") is a process where ZFS aims to store a single copy
+of any data that is repeated in different blocks or files, rather than storing 
+identical copies each time the data occurs. Deduplication 
+potentially allows repeated and identical data to be stored very 
+efficiently in a smaller space, but also places a 
+very heavy burden on RAM and on the CPU. In particular, dedup 
+significantly reduces write performance and usually requires large (and at 
+times huge) amounts of RAM.  See :ref:`Deduplication RAM requirements` for details.
+Because of the demands that dedup places on a system and the impact on performance, 
+enabling dedup is **usually not recommended** and the setting should be left in
+the default setting ("off") in most situations.
+
+.. warning::
+
+   The more data written to a deduplicated dataset, the more RAM it 
+   requires. The DDTs (dedup tables) also compete with other data for RAM,
+   and can be pushed out of RAM in some cases. When the system starts 
+   storing DDTs on disk because they no longer fit into RAM or have been
+   pushed out of RAM, performance craters.
+   
+   Importing an unclean deduplicated pool can also require large amounts of RAM, 
+   so that the deduplicated data in the pool can be checked during import. 
+   If the system does not have the needed RAM, it will panic. The only 
+   solution then is to add more RAM or recreate the pool.
+
+   **There is also no quick way to undedup data within a dataset once deduplication 
+   is enabled, since disabling deduplication DOES NOT CHANGE existing data.** The 
+   simplest way to undedup existing data is to copy the deduped data to a new dataset 
+   or location that does not have dedup set. The data will be undeduped when written. 
+   The simplest way to undedup data and keep it in the same pool, zvol or dataset is 
+   as follows: disable dedup on the pool, zvol or dataset, then copy the data within 
+   the same pool, zvol or dataset. Once the copy is confirmed to be correct, the 
+   original data can optionally be deleted. If the pool does not have enough free space
+   for this procedure, then disable dedup, copy the data to another system or storage
+   device, check the copies are correct and safe, then delete the original data, and 
+   transfer back the copied data to the pool.
+
```

Comment:
Remove "The simplest way to undedup" Start as a statement. "To remove deduplication but keep the same"
remove "as follows" and just give the procedure.
Rewrite this section with the above in mind.

Reply:
Rewritten as bullet list, to cover the different scenarios

Reply:
Done.
Also reformatted as a bullet list to cover the different scenarios